mozetl/search/aggregates.py

```diff
@@ -129,64 +186,85 @@ def add_derived_columns(exploded_search_counts):
     )
 
 
-def generate_dashboard(submission_date, bucket, prefix,
-                       input_bucket=DEFAULT_INPUT_BUCKET,
-                       input_prefix=DEFAULT_INPUT_PREFIX,
-                       save_mode=DEFAULT_SAVE_MODE):
-    start = datetime.datetime.now()
-    spark = (
-        SparkSession
-        .builder
-        .appName('search_dashboard_etl')
-        .getOrCreate()
-    )
+def gen_etl_function(transform_func, version):
+    """Generate and ETL job for a given Transform function
 
-    version = 3
-    source_path = 's3://{}/{}/submission_date_s3={}'.format(
-        input_bucket,
-        input_prefix,
-        submission_date
-    )
-    output_path = 's3://{}/{}/v{}/submission_date_s3={}'.format(
-        bucket,
-        prefix,
-        version,
-        submission_date
-    )
+    Takes a function to be applied to main_summary data and returns a function
+    which loads main_summary, executes the transform, and saves the results to
+    S3.
+    """
 
-    logger.info('Loading main_summary...')
-    main_summary = spark.read.parquet(source_path)
+    def etl_func(submission_date, bucket, prefix,
+                 input_bucket=DEFAULT_INPUT_BUCKET,
+                 input_prefix=DEFAULT_INPUT_PREFIX,
+                 save_mode=DEFAULT_SAVE_MODE):
+        """Load main_summary, apply transform_func, and write to S3"""
+        start = datetime.datetime.now()
+        spark = (
+            SparkSession
+            .builder
+            .appName('search_dashboard_etl')
+            .getOrCreate()
+        )
 
-    logger.info('Running the search dashboard ETL job...')
-    search_dashboard_data = search_aggregates(main_summary)
+        source_path = 's3://{}/{}/submission_date_s3={}'.format(
+            input_bucket,
+            input_prefix,
+            submission_date
+        )
+        output_path = 's3://{}/{}/v{}/submission_date_s3={}'.format(
+            bucket,
+            prefix,
+            version,
+            submission_date
+        )
 
-    logger.info('Saving rollups to: {}'.format(output_path))
-    (
-        search_dashboard_data
-        .repartition(10)
-        .write
-        .mode(save_mode)
-        .save(output_path)
-    )
+        logger.info('Loading main_summary...')
+        main_summary = spark.read.parquet(source_path)
+
+        logger.info('Running the search dashboard ETL job...')
```

Comment:
Could we pass in the name of the function here and make this log message reflect that?

Reply:
Done